insertion-sort.md

```diff
@@ -0,0 +1,100 @@
+# Insertion Sort
+
+Insertion sort is a comparision based sorting. In this sorting technique, we maintain a sub-list which is always sorted and then we take one element from the list and insert it at its correct place. We does so till all elements are inserted into sublist. For example, while playing cards we sort cards in our hand. Starting from left and moving to right,we keep on inserting the card at it's right place till end.
+
+## Example
+
+![alt tag](https://cloud.githubusercontent.com/assets/13117482/15633518/04ca4468-25cc-11e6-96af-feb395b456e0.png)
+
+In the above example, Grey Shaded sublist is always sorted and in each step we are inserting leftmost element of white sublist at it's correct position. Hence, we have sorted the complete list in this way.
+
+## Algorithm
+
+Loop for i=0 to N-1:
+* Pick element array[i] and insert it into sorted sublist array[0...i-1]
+
+## Complexity
+
+```
+Space complexity: O(1)      // Auxillary/temporary space is used.
+
+Time complexity: O(n*n)     // Two nested for loops are used.
+```
+
+## Cpp Implementation
+
+```cpp
+// Function to sort an array using insertion sort
+void insertionSort(int arr[], int n)
+{
+   int i, key, j;
+   for (i = 1; i < n; i++)
+   {
+       key = arr[i];
+       j = i-1;
+
+       /* Move elements of arr[0..i-1], that are greater than key,
+          to one position ahead of their current position */
+       while (j >= 0 && arr[j] > key)
+       {
+           arr[j+1] = arr[j];
+           j = j-1;
+       }
+       arr[j+1] = key; // place element key at it's correct place
+   }
+}
+
+int main()
+{
+    // array to be sorted
+    int arr[5] = {12, 11, 13, 5, 6};
+
+    // call the insertion sort
+    insertionSort(arr, 5);
+
+    // prints sorted array i.e. 5 6 11 12 13
+    for(int i=0; i<5; i++)
+        std::cout << arr[i] << " ";
+    return 0;
+}
+```
+
+:rocket: [Run Code](https://repl.it/CWZq)
```

Comment:
Add a newline at the end of printing. The cursor goes to next line which looks better.